apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts

```diff
@@ -198,23 +201,34 @@ export class DotEditContentHtmlService {
     }
 
     /**
-     * Render a contrentlet in the DOM after add it
+     * Render a contentlet in the DOM after add it
      *
-     * @param * contentlet
+     * @param DotPageContent contentlet
+     * @param string placeholderIdToBeReplaced
      * @memberof DotEditContentHtmlService
      */
-    renderAddedContentlet(contentlet: DotPageContent): void {
+    renderAddedContentlet(contentlet: DotPageContent, placeholderIdToBeReplaced?: string): void {
         const doc = this.getEditPageDocument();
+        if (placeholderIdToBeReplaced) {
+            const container: HTMLElement = doc.querySelector(`#${placeholderIdToBeReplaced}`).closest('[data-dot-object="container"]');
+            this.setContainterToAppendContentlet({ identifier: container.dataset['dotIdentifier'], uuid: container.dataset['dotUuid']});
+        }
+
         const containerEl: HTMLElement = doc.querySelector(
-            // eslint-disable-next-line max-len
             `[data-dot-object="container"][data-dot-identifier="${this.currentContainer.identifier}"][data-dot-uuid="${this.currentContainer.uuid}"]`
         );
 
         if (this.isContentExistInContainer(contentlet, containerEl)) {
             this.showContentAlreadyAddedError();
         } else {
-            const contentletPlaceholder = this.getContentletPlaceholder();
-            containerEl.appendChild(contentletPlaceholder);
+            let contentletPlaceholder;
```

Comment:
`contentletPlaceholder` is defined here and in line 285

Reply:
I don't see a valid reason why